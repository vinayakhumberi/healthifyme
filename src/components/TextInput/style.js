import styled from 'styled-components';
import { rem } from '../../utils/style.util';


export const Input = styled.input`
border: 1px solid;
width: ${props =>
    typeof props.width === 'string' ? props.width : rem(props.width)};
height: ${props =>
  typeof props.height === 'string' ? props.height : rem(props.height)};
color: ${props => props.color};
font-family: ${props => props.fontFamily};
font-size: ${props => rem(props.fontSize)};
font-weight: ${props => props.fontWeight};
font-style: ${props => props.fontStyle};
padding-left: ${props => rem(props.paddingLeft)};
padding-right: ${props => rem(props.paddingRight)};
padding-top: ${props => rem(props.paddingTop)};
padding-bottom: ${props => rem(props.paddingBottom)};
margin-top: ${props => (props.marginTop ? rem(props.marginTop) : "0px")};
margin-bottom: ${props =>
  props.marginBottom ? rem(props.marginBottom) : "0px"};
margin-right: ${props =>
  props.marginRight ? rem(props.marginRight) : "0px"};
margin-left: ${props => (props.marginLeft ? rem(props.marginLeft) : "0px")};
text-align: ${props => props.textAlign};
border-color: ${props => props.borderColor};
border-radius: 4px;
&:focus{
  outline: none;
}
`;