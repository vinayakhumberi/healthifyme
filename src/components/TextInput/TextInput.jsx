import React from 'react';
import PropTypes from "prop-types";

import { Input } from "./style";

const TextInput = function (props) {
  return (
    <Input
      data-test="text-input-container"
      autoComplete="off"
      type={props.elementType}
      color={props.color}
      fontFamily={props.fontFamily}
      fontSize={props.fontSize}
      fontWeight={props.fontWeight}
      fontStyle={props.fontStyle}
      paddingLeft={props.paddingLeft}
      paddingBottom={props.paddingBottom}
      paddingRight={props.paddingRight}
      paddingTop={props.paddingTop}
      textAlign={props.textAlign}
      lineHeight={props.lineHeight}
      letterSpacing={props.letterSpacing}
      marginLeft={props.marginLeft}
      marginBottom={props.marginBottom}
      marginRight={props.marginRight}
      marginTop={props.marginTop}
      width={props.width}
      height={props.height}
      borderColor={props.borderColor}
      placeholder={props.placeHolder}
      onChange={props.onChange}
      onBlur={props.onBlur}
    />
  );
};

TextInput.propTypes = {
  elementType: PropTypes.string,
  color: PropTypes.string,
  fontSize: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  fontWeight: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  fontStyle: PropTypes.string,
  paddingLeft: PropTypes.number,
  paddingRight: PropTypes.number,
  paddingTop: PropTypes.number,
  paddingBottom: PropTypes.number,
  text: PropTypes.string.isRequired,
  textAlign: PropTypes.string,
  lineHeight: PropTypes.number,
  letterSpacing: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  marginTop: PropTypes.number,
  marginBottom: PropTypes.number,
  marginLeft: PropTypes.number,
  marginRight: PropTypes.number,
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  borderColor: PropTypes.string,
  placeHolder: PropTypes.string,
  onChange: PropTypes.func,
  onBlur: PropTypes.func
};

TextInput.defaultProps = {
  color: "#343434",
  fontSize: 14,
  fontWeight: 400,
  fontStyle: "normal",
  fontFamily: 'sans-serif',
  textAlign: "initial",
  lineHeight: null,
  letterSpacing: "0.11px",
  text: "",
  paddingLeft: 0,
  paddingBottom: 0,
  paddingRight: 0,
  paddingTop: 0,
  marginLeft: 0,
  marginBottom: 0,
  marginRight: 0,
  marginTop: 0,
  elementType: "text",
  width: 200,
  height: 45,
  borderColor: '#eee',
  placeHolder: '',
  onChange: () => {},
  onBlur: () => {}
};

export default React.memo(TextInput);