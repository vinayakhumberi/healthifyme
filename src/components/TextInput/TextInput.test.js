import React from 'react';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import TextInput from './TextInput';

describe('<TextInput />', () => {
  let wrapper;
  let onInputChange = sinon.spy();
  let onInputBlur = sinon.spy();
  const text = "This is a test text";
  beforeEach(() => wrapper = shallow(<TextInput text={text} onChange={onInputChange} onBlur={onInputBlur} />));
  it('Should Render a TextInput Container', () => {
    expect(wrapper.find('[data-test="text-input-container"]').length).toEqual(1);
  });

  // Should find a way to do test onChange and onBlur
});


describe('<TextInput {...props} />', () => {
  let wrapper;
  const props = {
    color: "#343434",
    fontSize: 14,
    fontWeight: 400,
    fontStyle: "normal",
    fontFamily: 'sans-serif',
    textAlign: "initial",
    lineHeight: null,
    letterSpacing: "0.11px",
    paddingLeft: 0,
    paddingBottom: 0,
    paddingRight: 0,
    paddingTop: 0,
    marginLeft: 0,
    marginBottom: 0,
    marginRight: 0,
    marginTop: 0,
    width: 200,
    height: 45,
    borderColor: '#eee'
  };
  beforeEach(() => wrapper = shallow(<TextInput {...props} />));

  it('Should Render a TextInput Container', () => {
    expect(wrapper.find('[data-test="text-input-container"]').length).toEqual(1);
  });

  it('Should pass color prop to TextInput', () => {
    expect(wrapper.find('[data-test="text-input-container"]').get(0).props.color).toEqual(props.color);
  });
  it('Should pass fontSize prop to TextInput', () => {
    expect(wrapper.find('[data-test="text-input-container"]').get(0).props.fontSize).toEqual(props.fontSize);
  });
  it('Should pass fontWeight prop to TextInput', () => {
    expect(wrapper.find('[data-test="text-input-container"]').get(0).props.fontWeight).toEqual(props.fontWeight);
  });
  it('Should pass fontStyle prop to TextInput', () => {
    expect(wrapper.find('[data-test="text-input-container"]').get(0).props.fontStyle).toEqual(props.fontStyle);
  });
  it('Should pass fontFamily prop to TextInput', () => {
    expect(wrapper.find('[data-test="text-input-container"]').get(0).props.fontFamily).toEqual(props.fontFamily);
  });
  it('Should pass textAlign prop to TextInput', () => {
    expect(wrapper.find('[data-test="text-input-container"]').get(0).props.textAlign).toEqual(props.textAlign);
  });
  it('Should pass lineHeight prop to TextInput', () => {
    expect(wrapper.find('[data-test="text-input-container"]').get(0).props.lineHeight).toEqual(props.lineHeight);
  });
  it('Should pass letterSpacing prop to TextInput', () => {
    expect(wrapper.find('[data-test="text-input-container"]').get(0).props.letterSpacing).toEqual(props.letterSpacing);
  });
  it('Should pass paddingLeft prop to TextInput', () => {
    expect(wrapper.find('[data-test="text-input-container"]').get(0).props.paddingLeft).toEqual(props.paddingLeft);
  });
  it('Should pass paddingBottom prop to TextInput', () => {
    expect(wrapper.find('[data-test="text-input-container"]').get(0).props.paddingBottom).toEqual(props.paddingBottom);
  });
  it('Should pass paddingRight prop to TextInput', () => {
    expect(wrapper.find('[data-test="text-input-container"]').get(0).props.paddingRight).toEqual(props.paddingRight);
  });
  it('Should pass paddingTop prop to TextInput', () => {
    expect(wrapper.find('[data-test="text-input-container"]').get(0).props.paddingTop).toEqual(props.paddingTop);
  });
  it('Should pass marginLeft prop to TextInput', () => {
    expect(wrapper.find('[data-test="text-input-container"]').get(0).props.marginLeft).toEqual(props.marginLeft);
  });
  it('Should pass marginBottom prop to TextInput', () => {
    expect(wrapper.find('[data-test="text-input-container"]').get(0).props.marginBottom).toEqual(props.marginBottom);
  });
  it('Should pass marginRight prop to TextInput', () => {
    expect(wrapper.find('[data-test="text-input-container"]').get(0).props.marginRight).toEqual(props.marginRight);
  });
  it('Should pass marginTop prop to TextInput', () => {
    expect(wrapper.find('[data-test="text-input-container"]').get(0).props.marginTop).toEqual(props.marginTop);
  });
  it('Should pass elementType prop to TextInput', () => {
    expect(wrapper.find('[data-test="text-input-container"]').get(0).props.elementType).toEqual(props.elementType);
  });
  it('Should pass width prop to TextInput', () => {
    expect(wrapper.find('[data-test="text-input-container"]').get(0).props.width).toEqual(props.width);
  });
  it('Should pass height prop to TextInput', () => {
    expect(wrapper.find('[data-test="text-input-container"]').get(0).props.height).toEqual(props.height);
  });
  it('Should pass borderColor prop to TextInput', () => {
    expect(wrapper.find('[data-test="text-input-container"]').get(0).props.borderColor).toEqual(props.borderColor);
  });
});

