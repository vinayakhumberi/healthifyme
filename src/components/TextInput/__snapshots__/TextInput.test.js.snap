// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> renders correctly 1`] = `ShallowWrapper {}`;

exports[`<TextInput {...props} /> renders correctly 1`] = `ShallowWrapper {}`;
